src/lti_toolbox/lti.py

```diff
@@ -33,37 +29,47 @@ def __init__(self, request):
         self._params = {}
         self._verified = False
 
+    def _process_params(self) -> Union[SelectionParams, LaunchParams]:
+        """Process LTI parameters based on request type."""
+
+        try:
+            is_selection_request = (
+                self.request.POST.get("lti_message_type")
+                == SELECTION_PARAMS_MESSAGE_TYPE
+            )
+
+            if is_selection_request:
+                return SelectionParams(self.request.POST)
+
+            return LaunchParams(self.request.POST)
+
+        except ParamException as error:
+            raise LTIException(f"Exception while processing parameters: {error}")
```

Comment:
You should reduce to its minimum the scope of try/except blocks to make crystal clear what you are trying to catch:

```suggestion
        is_selection_request = (
            self.request.POST.get("lti_message_type")
            == SELECTION_PARAMS_MESSAGE_TYPE
        )

        try:
            params = SelectionParams(self.request.POST) if is_selection_request else LaunchParams(self.request.POST)
        except ParamException as error:
            raise LTIException(f"Exception while processing parameters: {error}")

        return params
```

Reply:
Improved !